### PR TITLE
Throw when receiver is passed in TimeSpan.Zero for server timeout

### PR DESCRIPTION
**Pull request by SeanFeldman**
## Description

Initial stab at #294

### CODE REVIEW

**Comment by SeanFeldman** [comment]
Compilation fails on the build server, but locally is fine... not sure why.

**Comment by nemakam** [comment]
@SeanFeldman, could you take a look at the build failure?

**Comment by SeanFeldman** [comment]
@nemakam will do. Though I have to admit I have no clue why it is failing on the build server 😕

**Comment by nemakam** [comment]
Okay. Let me also give it a shot..

File: src/Microsoft.Azure.ServiceBus/Resources.Designer.cs [hunk 1]
```diff
@@ -10,47 +10,48 @@
...
-        ///    Returns the cached ResourceManager instance used by this class.
+        ///   Returns the cached ResourceManager instance used by this class.
         /// </summary>
         [global::System.ComponentModel.EditorBrowsableAttribute(global::System.ComponentModel.EditorBrowsableState.Advanced)]
-        public static global::System.Resources.ResourceManager ResourceManager {
+        internal static global::System.Resources.ResourceManager ResourceManager {
             get {
                 if (object.ReferenceEquals(resourceMan, null)) {
-                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.Azure.ServiceBus.Resources", typeof(Resources).GetTypeInfo().Assembly);
+                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.Azure.ServiceBus.Resources", typeof(Resources).Assembly);
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Resources.Designer.cs]: Try reverting this change. Was this a manual change or automated change?

**Comment by SeanFeldman** [comment]
@nemakam, don't bother. I'm an idiot. Fixed.

**Comment by SeanFeldman** [comment]
Two tests failing on the build server:

1. `RetryTests.RunOperationShouldReturnImmediatelyIfRetryIntervalIsGreaterThanOperationTimeout`
1. `WebSocketsEnd2EndTests.SendAndReceiveWithWebSockets`

The first one is passing locally
image 

The second one is has been failing in [`dev` branch](https://ci.appveyor.com/project/vinaysurya/azure-service-bus-dotnet/branch/dev/job/th03ddpnqc5rf7my/tests).

**Comment by SeanFeldman** [comment]
@nemakam I can't trigger another build to validate the flaky test (first one). Can you please?

**Comment by SeanFeldman** [comment]
The build is broken due to the WebSockets related test that is already broken in `dev` 😞 
image

Unless there's anything else, this can be merged.

**Comment by SeanFeldman** [comment]
@nemakam could you please confirm what I've commented [here](https://github.com/Azure/azure-service-bus-dotnet/pull/296#issuecomment-328016992)?

**Comment by nemakam** [comment]
@SeanFeldman, Yes. WebSocket test has been failing since the time it was checked in.
#299 should take care of that. I've merged those changes with your branch which automatically triggered a new build. Lets wait for this one

**Pull request merged**